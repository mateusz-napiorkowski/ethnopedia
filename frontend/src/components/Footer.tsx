import React from "react";
import PPLogo from '../assets/images/PP_znak_konturowy_CMYK.svg';
import KPO from '../assets/images/Znak_KPO_Poziom_RGB.png.png';
import RP from '../assets/images/Znak_barw_RP_linia_zamykajaca_Poziom_RGB.png';
import NGEU from '../assets/images/Znak_NextGenerationEU_Poziom_RGB.png.png';
import Dariah_PL from '../assets/images/logo-dariah.png';
import ISPAN from '../assets/images/Logotyp_ISPAN.png';

const Footer = () => {
    return (
        <section className="bg-white dark:bg-gray-800 py-10 w-full">
            <div
                className="max-w-screen-xl mx-auto px-4 lg:px-12 text-center text-gray-600 dark:text-gray-300 space-y-10">

                {/* Tekst o projekcie */}
                <div className="text-sm">
                    <p className="mb-2">
                        Narzędzie jest realizowane w ramach projektu{" "}
                        <a
                            href="https://www.dariah.pl"
                            className="text-primary-500 hover:text-primary-700"
                        >
                            Cyfrowa Infrastruktura Badawcza dla Humanistyki i Nauk o Sztuce DARIAH-PL
                        </a>{" "}
                        finansowanego z funduszy Krajowego Planu Odbudowy (KPOD.01.18-IW.03-0013/23) przez
                        Politechnikę Poznańską we współpracy z Instytutem Sztuki PAN.
                    </p>
                </div>

                {/* Główne logotypy */}
                <div className="flex flex-wrap justify-center items-center gap-16">
                    <img src={KPO} alt="Logo Krajowego Planu Odbudowy" className="h-20"/>
                    <img src={RP} alt="Logo Barw RP" className="h-20"/>
                    <img src={NGEU} alt="Logo NextGenerationEU" className="h-20"/>
                </div>
                <div className="flex flex-wrap justify-center items-center gap-16">
                    <img src={Dariah_PL} alt="Logo Dariah-PL" className="h-14"/>
                    <img src={PPLogo} alt="Logo Politechniki Poznańskiej" className="h-28"/>
                    <img src={ISPAN} alt="Logo Instytutu Sztuki Polskiej Akademi Nauk" className="h-10"/>
                </div>

                {/* Dolna sekcja */}
                <div className="text-sm text-gray-500 dark:text-gray-400 space-y-1">
                    <p>
                        &copy; {new Date().getFullYear()} Politechnika Poznańska – MusicPUT Team
                    </p>
                    <p>
                        Kontakt:{" "}
                        <a
                            href="mailto:dariah@cs.put.poznan.pl"
                            className="text-primary-500 hover:text-primary-700"
                        >
                            dariah@cs.put.poznan.pl
                        </a>
                    </p>
                </div>
            </div>
        </section>

    );
};

export default Footer;
