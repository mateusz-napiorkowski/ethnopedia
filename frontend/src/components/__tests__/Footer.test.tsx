import '@testing-library/jest-dom';
import { render, } from '@testing-library/react'
import Footer from "../Footer"
import { MemoryRouter, Routes, Route } from "react-router-dom";


const renderComponent = () => {
    return render(
            <MemoryRouter initialEntries={[`/`]}>
                <Routes>
                    <Route path="/" element={<Footer/>}/>
                </Routes>  
            </MemoryRouter>
        
    );
};

describe("Footer tests", () => {
    beforeEach(() => {
        jest.clearAllMocks()
    });

    it("should render footer component", () => {
        const {container} = renderComponent()
        expect(container).toMatchSnapshot()
    })
})