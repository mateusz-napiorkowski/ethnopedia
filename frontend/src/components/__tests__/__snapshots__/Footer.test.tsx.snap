// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer tests should render footer component 1`] = `
<div>
  <section
    class="bg-white dark:bg-gray-800 py-10 w-full"
  >
    <div
      class="max-w-screen-xl mx-auto px-4 lg:px-12 text-center text-gray-600 dark:text-gray-300 space-y-10"
    >
      <div
        class="text-sm"
      >
        <p
          class="mb-2"
        >
          Narzędzie jest realizowane w ramach projektu
           
          <a
            class="text-primary-500 hover:text-primary-700"
            href="https://www.dariah.pl"
          >
            Cyfrowa Infrastruktura Badawcza dla Humanistyki i Nauk o Sztuce DARIAH-PL
          </a>
           
          finansowanego z funduszy Krajowego Planu Odbudowy (KPOD.01.18-IW.03-0013/23) przez Politechnikę Poznańską we współpracy z Instytutem Sztuki PAN.
        </p>
      </div>
      <div
        class="flex flex-wrap justify-center items-center gap-16"
      >
        <img
          alt="Logo Krajowego Planu Odbudowy"
          class="h-20"
          src="test-file-stub"
        />
        <img
          alt="Logo Barw RP"
          class="h-20"
          src="test-file-stub"
        />
        <img
          alt="Logo NextGenerationEU"
          class="h-20"
          src="test-file-stub"
        />
      </div>
      <div
        class="flex flex-wrap justify-center items-center gap-16"
      >
        <img
          alt="Logo Dariah-PL"
          class="h-14"
          src="test-file-stub"
        />
        <img
          alt="Logo Politechniki Poznańskiej"
          class="h-28"
          src="[object Object]"
        />
        <img
          alt="Logo Instytutu Sztuki Polskiej Akademi Nauk"
          class="h-10"
          src="test-file-stub"
        />
      </div>
      <div
        class="text-sm text-gray-500 dark:text-gray-400 space-y-1"
      >
        <p>
          © 
          2025
           Politechnika Poznańska – MusicPUT Team
        </p>
        <p>
          Kontakt:
           
          <a
            class="text-primary-500 hover:text-primary-700"
            href="mailto:dariah@cs.put.poznan.pl"
          >
            dariah@cs.put.poznan.pl
          </a>
        </p>
      </div>
    </div>
  </section>
</div>
`;
